TemperatureMeter.c

```diff
@@ -0,0 +1,129 @@
+/*
+htop - TemperatureMeter.c
+(C) 2013 Ralf Stemmer
+(C) 2014 Blair Bonnett
+(C) 2020 Maxim Kurnosenko
+Released under the GNU GPL, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "TemperatureMeter.h"
+
+#include "ProcessList.h"
+#include "CRT.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+#include <errno.h>
+#include <sensors/sensors.h>
+
+/*{
+#include "Meter.h"
+}*/
+
+static int get_cpu_temp(double *temp) {
+    int n = 0;
+
+    for (const sensors_chip_name *chip = sensors_get_detected_chips(NULL, &n); chip; chip = sensors_get_detected_chips(NULL, &n)) {
+        int r;
+        char buffer[256];
+        r = sensors_snprintf_chip_name(buffer, sizeof(buffer), chip);
+        if (r < 0)
+            continue;
+
+        if (strncmp(buffer, "coretemp", strlen("coretemp")) != 0)
+            continue;
+
+        int m = 0;
+        for (const sensors_feature *feature = sensors_get_features(chip, &m); feature; feature = sensors_get_features(chip, &m)) {
+            if (feature->type != SENSORS_FEATURE_TEMP)
+                continue;
+
+            if (feature->number != 0)
+                continue;
+
+            const sensors_subfeature *sub_feature = sensors_get_subfeature(chip, feature, SENSORS_SUBFEATURE_TEMP_INPUT);
+            if (sub_feature) {
+                double local_temp;
+                r = sensors_get_value(chip, sub_feature->number, &local_temp);
+                if (r == 0) {
+                    *temp = local_temp;
+                    return 0;
+                }
+            }
+        }
+    }
+
+    return -ENOENT;
+}
+
+int TemperatureMeter_attributes[] = {
+   TEMPERATURE_COOL,
+   TEMPERATURE_MEDIUM,
+   TEMPERATURE_HOT,
+};
+
+static void TemperatureMeter_setValues(Meter* this, char* buffer, int len) {
+   ProcessList* pl = this->pl;
+   this->total = pl->totalTasks;
+   this->values[0] = pl->runningTasks;
+   snprintf(buffer, len, "%d/%d", (int) this->values[0], (int) this->total);
+}
+
+static void TemperatureMeter_display(Object* cast, RichString* out) {
+   (void) cast;
+   int r, temperature;
+   double temp;
+
+   r = sensors_init(NULL);
```

Comment:
Do not call `sensors_init`+`sensors_cleanup` every second.
Call it once in `htop.c` or some other top-level code.